src/gConsent/guard/isDatasetCore.ts

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import type { DatasetCore } from "@rdfjs/types";
+
+/**
+ * Auxiliary function to determine whether the candidateDataset implements the DatasetCore
+ * interface.
+ * @param candidateDataset the candidate to be checked.
+ * @returns true if the candidate implements the DatasetCore interface, false otherwise.
+ * @internal
+ */
+function isDatasetCore(
+  candidateDataset: unknown | DatasetCore,
+): candidateDataset is DatasetCore {
+  return typeof (candidateDataset as DatasetCore).add === "function";
+}
+
+export { isDatasetCore };
```

Comment:
```suggestion
export function isDatasetCore(
  candidateDataset: unknown | DatasetCore,
): candidateDataset is DatasetCore {
  return typeof (candidateDataset as DatasetCore).add === "function";
}
```
The export declarations are inlined in the rest of the codebase.